crates/sign/src/lib.rs

```diff
@@ -56,6 +56,18 @@ fn verify_message(msg: &Message, signature: &Signature) -> Result<(), secp256k1:
     secp.verify_ecdsa(msg, &sig, &pk)
 }
 
+/// Verify the given [`PublicKey`] against the message & signature.
+pub fn verify_signature(
```

Comment:
Looks like we have a `verify_message` that is the same but recovers the public key first.

Maybe we should:

1. name this `verify_message_with_pubkey` or similar
2. change `verify_message` to be implemented in terms of `verify_message_with_pubkey`

Reply:
yep, sounds like good changes to me. Will implement that now.